### PR TITLE
utm: 4.1.6 -> 4.2.5

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by rrbutani — COMMENTED** [review]
Thanks! I've been meaning to add an update script but keep forgetting.

At a glance this looks fine; will test later today once I have access to a macOS machine again (no need to block merge on this though).

File: pkgs/os-specific/darwin/utm/default.nix [hunk 1]
```diff
@@ -33,6 +34,11 @@ stdenvNoCC.mkDerivation rec {
     runHook postInstall
   '';
 
+  passthru.updateScript = gitUpdater {
+    url = "https://github.com/utmapp/UTM.git";
+    rev-prefix = "v";
+  };
+
```
Comment (rrbutani) [inline pkgs/os-specific/darwin/utm/default.nix:41]: My concern with using the `gitUpdater` directly is that UTM tends to push tags for minor releases that are then marked as "pre-release"; i.e. `4.2.4` was the first stable `4.2.x` release: https://github.com/utmapp/UTM/releases

I'm not familiar with how the updater script is actually triggered by r-ryantm; repology distinguishes between stable/pre-release tags already so maybe this isn't an issue?

Comment (wegank) [reply pkgs/os-specific/darwin/utm/default.nix:41]: I don't think `list-git-tags` distinguishes between `4.2.3` and `4.2.4`, so that will be problematic.

Comment (wegank) [reply pkgs/os-specific/darwin/utm/default.nix]: Using `nix-update-script` instead.

Comment (rrbutani) [reply pkgs/os-specific/darwin/utm/default.nix]: Does `nix-update-script` distinguish pre-releases?

Maybe I'm missing something but it [doesn't look like it does](https://github.com/Mic92/nix-update/blob/6639ad481c667be11b50fd482b30b4702922df63/nix_update/version/github.py#L13-L36); the Atom feeds it parses ([UTM's as an example](https://github.com/utmapp/UTM/releases.atom)) don't seem to encode this information.

Comment (wegank) [reply pkgs/os-specific/darwin/utm/default.nix]: Probably worth promoting the expression to an general updater.

**Pull request merged**